### ISSUE TITLE
theme styles generate duplication css output

### ISSUE THREAD

**Issue by zhaohuanyuu** [system-ui/theme-ui#2349]
**Describe the bug**
When I want to use the styles attribute to customize the native html tag style, for example: blockquote, but the style that ends up being generated to the page is repeated twice

**To Reproduce**
1. clone my reproduction project [blog production](https://github.com/zhaohuanyuu/blog-reproduction/tree/main)
2. yarn install & yarn start
3. Click the Blog header
4. Route to the named reproduction article
5. Inspect text content is "this is theme-ui style custom style" element

**Expected behavior**

I don't know why generate duplication css output,that should generate css only once

**Screenshots**
![image](https://user-images.githubusercontent.com/25791574/197392034-62304b2a-2006-4d92-b88f-071c51e8741f.png)

![image](https://user-images.githubusercontent.com/25791574/197390105-f2170d86-de68-453c-ac33-d31f1511e8b8.png)

**Additional context**
As long as the styles defined from the styles object of theme-ui config have the problem of generating duplicate css

**Comment by lachlanjc** [comment]
Yep, I'm having this issue too on sites like https://lachlanjc.com/ — not sure what's going on

**Comment by hasparus** [comment]
I don't see this problem in a commercial project nor in [CodeSandbox](https://codesandbox.io/s/theme-ui-0-15-0-develop-test-ovttsl?file=/next.config.js).

Also, not all classes in @lachlanjc's websites have duplicate styles.

<img width="336" alt="image" src="https://user-images.githubusercontent.com/15332326/202686771-3aa8ae5a-0e04-4a97-85bb-b17474914b70.png">

Do you guys think we can narrow this? When does it happen? When does it not happen?

**Comment by mbektimirov** [comment]
I can confirm that duplication takes place in some undisclosed cases. It's not clear at the moment what causes this behavior. 

<img width="429" alt="Screenshot 2023-02-14 at 18 52 33" src="https://user-images.githubusercontent.com/918009/218773447-96b171b8-d3ee-493f-90d8-e762fe2095c6.png">

**Comment by simeng-li** [comment]
@hasparus wondering if there are any updates for this issue. I am still able to see this duplicate style issue on my new Gatsby project.   Using the `gatsby-plugin-theme-ui `plugin.

**Comment by techfg** [comment]
> I don't see this problem in a commercial project nor in [CodeSandbox](https://codesandbox.io/s/theme-ui-0-15-0-develop-test-ovttsl?file=/next.config.js).
> 
> Also, not all classes in @lachlanjc's websites have duplicate styles.
> 
> <img alt="image" width="336" src="https://user-images.githubusercontent.com/15332326/202686771-3aa8ae5a-0e04-4a97-85bb-b17474914b70.png">
> 
> Do you guys think we can narrow this? When does it happen? When does it not happen?

@hasparus The problem appears to be specific to Gatsby and changes introduced in the Head API in Gastby 5.6.  See [Gatsby #38249](https://github.com/gatsbyjs/gatsby/issues/38249) for more details and repros at:

1. https://github.com/techfg/gatsby-duplicate-styles-when-using-head-api.git
2. https://codesandbox.io/p/sandbox/duplicate-style-element-in-head-when-using-head-api-and-theme-ui-h8zqtk
